receiver/mongodbreceiver/factory.go

```diff
@@ -28,32 +28,28 @@ import (
 	"github.com/GoogleCloudPlatform/opentelemetry-operations-collector/receiver/mongodbreceiver/internal/metadata"
 )
 
-const (
-	typeStr   = "mongodb"
-	stability = component.StabilityLevelBeta
-)
-
 // NewFactory creates a factory for mongodb receiver.
 func NewFactory() receiver.Factory {
 	return receiver.NewFactory(
-		typeStr,
+		metadata.Type,
 		createDefaultConfig,
-		receiver.WithMetrics(createMetricsReceiver, stability))
+		receiver.WithMetrics(createMetricsReceiver, metadata.MetricsStability))
 }
 
 func createDefaultConfig() component.Config {
 	return &Config{
-		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+		ControllerConfig: scraperhelper.ControllerConfig{
 			CollectionInterval: time.Minute,
 		},
 		Timeout: time.Minute,
-		Hosts: []confignet.NetAddr{
+		Hosts: []confignet.AddrConfig{
 			{
-				Endpoint: "localhost:27017",
+				Endpoint:  "localhost:27017",
+				Transport: confignet.TransportTypeTCP,
```

Comment:
Copies how upstream updated this: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/mongodbreceiver/factory.go#L35